agent/reporter_test.go

```diff
@@ -21,17 +21,25 @@ func TestReporter(t *testing.T) {
 	const ntasks = 100
 
 	var (
-		ctx      = context.Background()
-		statuses = make(map[string]*api.TaskStatus) // destination map
-		unique   = make(map[uniqueStatus]struct{})  // ensure we don't receive any status twice
-		mu       sync.Mutex
-		expected = make(map[string]*api.TaskStatus)
-		wg       sync.WaitGroup
+		// NOTE(dperny): don't increase this timeout! this test should never
+		// take anything close to a minute. If it does, you have a bug
+		// someplace else. Probably a deadlock.
+		ctx, cancel = context.WithTimeout(context.Background(), 60*time.Second)
+		statuses    = make(map[string]*api.TaskStatus) // destination map
+		unique      = make(map[uniqueStatus]struct{})  // ensure we don't receive any status twice
+		mu          sync.Mutex
+		expected    = make(map[string]*api.TaskStatus)
+		// this test used to use a waitgroup, but i've removed it because if
```

Comment:
Non-blocking nitpick:  this is an awesome comment to reviewers, but I'm not sure it is a useful comment in the code to say what the code used to do.  Perhaps just a comment at the top of the test saying that the failure condition is a deadlock?